lib/course_training_progress_manager.rb

```diff
@@ -110,9 +110,11 @@ def completed_exercise_modules_for_user_and_course
     TrainingModulesUsers
       .where(user_id: @user.id)
       .where(training_module_id: exercise_modules_for_course)
-      .where(flags: { marked_complete: true })
       .count do |tmu|
-        TrainingModule.find(tmu.training_module_id).exercise?
+        course_flags_hash = tmu.flags[@course.id] || tmu.flags
+        if course_flags_hash[:marked_complete] == true
+          TrainingModule.find(tmu.training_module_id).exercise?
+        end
```

Comment:
This does seem to solve the `:marked_complete` issue! Well done.  :)

I would ask for a couple changes to this though:

1. We should query our database as little as possible. Instead of using `.find()` twice, we should store the value of that result in a variable. We also can probably get to the training module _through_ the `tmu` variable (e.g. `tmu.training_module`

2. We should be more explicit with returning `true` and `false`. The else case for the `if` conditional is implicit as opposed to explicit. To put that another way, we should feel confident about what value is actually being returned. In this case, we're either returning the result of `.exercise?` or... whatever is stored inside of `course_flags_hash`?

Great work, overall!

Reply:
Thanks @bwreid ! I have made the changes, please have a look.